src/javascript/binary/pages/portfiliows.js

```diff
@@ -0,0 +1,159 @@
+/** 
+ * mising data in api portfolio object:
+ *  - recomputed_price
+ *  - sell_channel
+ *  - shortcode
+ *  - ref number
+ *  - indicative amount      
+**/
+
+var PortfolioWS =  (function() {
+
+    'use strict';
+
+    var pageLanguage = page.language();
+    if(!pageLanguage) pageLanguage = "EN";
+
+    var init = function() {
+        TradeSocket.init({"thendo":"updateBalance"});
+    };
+
+    
+    var updateBallance = function(data) {
+        $("span[data-id='balance']").text(fixCurrency(data.authorize.balance, data.authorize.currency)).removeClass("loading");
+        if(0 === parseFloat(data.authorize.balance, 10)) {
+            $("#if-balance-zero").removeClass("dynamic");
+        } else {
+            $("#if-balance-zero").remove();
+        }
+    };
+
+    /**
+     * Updates portfolio table
+    **/
+    var updatePortfolio = function(data) {
+
+        /**
+         * Check for error
+        **/
+        if("error" in data) {
+            throw new Error("Trying to get portfolio data, we got this error", data.error);
+        }
+
+        /**
+         * no contracts
+        **/
+        if(0 === data.portfolio.contracts.length) {
+            $("#portfolio-table").remove();
+            $("#portfolio-no-contract").removeClass("dynamic");
+            return true;
+        }
+
+        /**
+         * User has at least one contract
+        **/
+
+        $("#portfolio-no-contract").remove();
+
+        var rowTemplate = '<div class="grd-parent grd-grid-12 grd-row-padding table-body table-body-lines"><div class="grd-grid-2 grd-hide-mobile grd-grid-phablet-2">!ref-number!</div><div class="grd-grid-5 grd-grid-mobile-7 grd-grid-phablet-6">!longcode!</div><div class="grd-grid-5 grd-grid-mobile-5 grd-grid-phablet-4"><div class="grd-grid-4 grd-grid-mobile-12 grd-grid-phablet-12 right-aligned">!currency! <strong>!buy_price!</strong></div><div class="grd-hide grd-show-mobile grd-grid-mobile-12 grd-show-phablet grd-grid-phablet-12"><br>Indicative :</div><div class="grd-grid-4 grd-grid-mobile-12 grd-grid-phablet-12 right-aligned">!currency! 0.00</div><div class="grd-hide grd-show-mobile grd-grid-mobile-12 grd-show-phablet grd-grid-phablet-12"><br></div><div class="grd-grid-4 grd-grid-mobile-12 grd-grid-phablet-12 center-aligned"><div><span class="button"><button class="button open_contract_details" onclick="return false;" contract_id="5266199068" controller_action="sell" currency="!currency!" l="!lang-code!" payout="!payout!" purchase_price="!buy_price!" purchase_time="2015-10-02 08:24:13" qty="1" recomputed_price="0" sell_channel="P_!contract_type!-1443774253_!symbol!_1444424400_14958600_14065600_c_!currency!_!lang-code!" shortcode="!contract_type!_!symbol!_100_1443774253_1444424400F_14958600_14065600" url="https://www.binary.com/trade/analyse_contract?l=!lang-code!">:view:</button></span></div></div></div></div>';
```

Comment:
Removed hard-coded values and used call to proposal_open_contract to get the required values in the most recent commits.